src/Mvc/src/Microsoft.AspNetCore.Mvc.Core/Resources.resx

```diff
@@ -493,4 +493,12 @@
   <data name="ReferenceToNewtonsoftJsonRequired" xml:space="preserve">
     <value>'{0}' requires a reference to '{1}'. Configure your application by adding a reference to the '{1}' package and calling '{2}.{3}' inside the call to '{4}' in the application startup code.</value>
   </data>
+  <data name="ModelBinding_ExceededMaxModelBindingCollectionSize" xml:space="preserve">
+    <value>Collection bound to '{0}' exceeded {1}.{2} ({3}). Remove properties that bind unconditionally from {4}.</value>
+    <comment>{0} is model name, {1} is MvcOptions, {2} is MaxModelBindingCollectionSize, {3} is option value, {4} is affected type.</comment>
+  </data>
+  <data name="ModelBinding_ExceededMaxModelBindingRecursionDepth" xml:space="preserve">
+    <value>Model binding system exceeded {0}.{1} ({2}). Reduce the potential nesting of {3}.</value>
+    <comment>{0} is MvcOptions, {1} is MaxModelBindingRecursionDepth, {2} is option value, {3} is (loopy or deeply nested) top-level model type.</comment>
+  </data>
```

Comment:
These error message should call out that you should go to `MvcOptions.xyz` and understand this more. I know you mention the existing of these properties, but there's good documentation there that we can refer to.